templates/shared/combomarkdowneditor.tmpl

```diff
@@ -41,12 +41,23 @@ Template Attributes:
 			</div>
 			<div class="markdown-toolbar-group gt-f1"></div>
 			<div class="markdown-toolbar-group">
-				<span class="markdown-toolbar-button markdown-switch-easymde">{{svg "octicon-arrow-switch"}}</span>
+				<button class="markdown-toolbar-button markdown-switch-monospace" role="switch" data-enable-text="{{.locale.Tr "editor.enable_monospace_font"}}" data-disable-text="{{.locale.Tr "editor.disable_monospace_font"}}">{{svg "octicon-typography"}}</button>
+				<button class="markdown-toolbar-button markdown-switch-easymde">{{svg "octicon-arrow-switch"}}</button>
 			</div>
 		</markdown-toolbar>
 		<text-expander keys=": @">
 			<textarea class="markdown-text-editor js-quick-submit"{{if .TextareaName}} name="{{.TextareaName}}"{{end}}{{if .TextareaPlaceholder}} placeholder="{{.TextareaPlaceholder}}"{{end}}{{if .TextareaAriaLabel}} aria-label="{{.TextareaAriaLabel}}"{{end}}>{{.TextareaContent}}</textarea>
 		</text-expander>
+		<script>
+			const monospaceEnabled = localStorage?.getItem('editor-monospace') === 'true';
+			const btn = document.querySelector('.markdown-switch-monospace');
+			const text = btn.getAttribute(monospaceEnabled ? 'data-disable-text' : 'data-enable-text');
+			btn.setAttribute('data-tooltip-content', text);
+			btn.setAttribute('aria-checked', String(monospaceEnabled));
+			if (monospaceEnabled) {
+				document.querySelector('.markdown-text-editor').classList.add('gt-mono');
+			}
+		</script>
```

Comment:
Why an inline script?
Why not add another `init` method to `index.js`?

Reply:
Which you already seem to be doing below?
Now I'm especially confused…

Reply:
Inline script is required to synchronously set correct textarea class, otherwise it will flash on page load until `index.js` is parsed and executed.

This is very similar to the file tree on diff, we can't delay setting the state, otherwise there will be wrong flashes.

Reply:
Inline script is unavoidable, otherwise there is a flash of wrong content.